config/config_processor.go

```diff
@@ -12,15 +13,71 @@ import (
 
 var ErrNotImplemented = errors.New("operation not implemented")
 
+// OutputOptionConfig structure for output file.
 type OutputOptionConfig op.EncoderOption
 
+// Config structure for output file.
+//
+// Format: Output file format. Either `jpeg` or `png`.
+//
+// NameSuffix: Suffix of the output file name.
+//
+// NamePrefix: Prefix of the output file name.
+//
+// Options: Encoder option. For jpeg use and supports only `Quality` option.
 type OutputConfig struct {
 	Format     string              `yaml:"format"`     // Output file format
 	NameSuffix string              `yaml:"nameSuffix"` // Output file name suffix
 	NamePrefix string              `yaml:"namePrefix"` // Output file name prefix
 	Options    *OutputOptionConfig `yaml:"options"`    // Encoder option
 }
 
+// Config structure for resizing image.
+//
+// Width: Output image width.
+//
+// Height: Output image height.
+//
+// Factor: Resize factor.
+//
+// Algorithm: Resize algorithm. Either `nearestneighbor`, `catmullrom`, or `approxbilinear`.
+//
+// NOTE: The `Factor` is prioritized over `Width` and `Height`.
+type ResizeConfig struct {
+	Width     int     `yaml:"width"`     // Output image width
+	Height    int     `yaml:"height"`    // Output image height
+	Factor    float32 `yaml:"factor"`    // Resize factor
+	Algorithm string  `yaml:"algorithm"` // Resize algorithm
```

Comment:
Could you tell me if there's a definition for the algorithm?

Reply:
Yes, there are 3 kinds of interpolation algorithm implemented in GoLang.
- `catmullrom`: Provides best quality.
- `Bilinear`: Familiar name, provides balanced quality and speed.
- `nearest neighborhood`: Your art will become pixel art.